### PR TITLE
Feature/extract params2

### PR DESCRIPTION
**Pull request by EstrellaFG**
New parameters added to **protocol_tomo_extraction**:

-`Box size` can be now introduced by the user. There is also a wizard (`EmanTomoExtractionWizard`) that selects automatically the same box size as picking.
-`Downsampling factor`: a factor chosen by the user to downsample the subtomograms after the extraction.
-The parameter `factor` (_cshrink_) is no longer introduced by the user, instead, it is computed automatically from the sampling rate of input data.

-Change _micrographs_ by _tomograms_ in comments, summary and methods.

For the correct working of `protocol_tomo_extraction` it is necessary to have the changes in the PR scipion-em/scipion-em-tomo/pull/18 of `scipion-em-tomo` repository

### CODE REVIEW

File: eman2/protocols/protocol_tomo_extraction.py [hunk 1]
```diff
@@ -146,42 +144,50 @@ def readSetOfTomograms(self, workDir, tomogramsSet):
...
         out.close()
 
     # --------------------------- STEPS functions -----------------------------
     def extractParticles(self):
-
-        args = '%s --coords %s --boxsize %d' % (
-            self.getInputTomogram().getFileName(), pwutils.replaceBaseExt(self.getInputTomogram().getFileName(), 'coords'),
-            self.inputCoordinates.get().getBoxSize())
-
+        TsTomo = self.getInputTomogram().getSamplingRate()
```
Comment (adrianq) [inline eman2/protocols/protocol_tomo_extraction.py]: This is a bit picky but not sure if TsTomo and TsCoord are particularly descriptive and in general variables should start with lower case

File: eman2/wizards.py [hunk 2]
```diff
@@ -118,3 +119,19 @@ def show(self, form):
         if myprops['applyChanges'] == 'true':
             for param in params:
                 form.setVar(param, myprops[param + '.value'])
+
+
+class EmanTomoExtractionWizard(EmWizard):
+    _targets = [(EmanProtTomoExtraction, ['boxSize'])]
+
+    def show(self, form):
+        tomoextractProt = form.protocol
```
Comment (adrianq) [inline eman2/wizards.py]: tomoExtractProt?

File: eman2/protocols/protocol_tomo_extraction.py (hunk 1 above)
Comment (adrianq) [reply eman2/protocols/protocol_tomo_extraction.py]: Maybe also a comment regarding this logic with TsTomo and TsCoord can be helpful...

Comment (EstrellaFG) [reply eman2/protocols/protocol_tomo_extraction.py]: solved in https://github.com/scipion-em/scipion-em-eman2/pull/33/commits/9c158b38a1db38591511411de61b3ca96e084186

File: eman2/wizards.py (hunk 2 above)
Comment (EstrellaFG) [reply eman2/wizards.py]: solved in https://github.com/scipion-em/scipion-em-eman2/pull/33/commits/9c158b38a1db38591511411de61b3ca96e084186

**Comment by EstrellaFG** [comment]
> @EstrellaFG Some minor comments here... Maybe you can also adapt the current tests (or create a new one if needed) to test the new params/logic

Tests modified in https://github.com/scipion-em/scipion-em-eman2/pull/33/commits/1357691bad688a23c1cf79e94056bd1538f9642c

File: eman2/protocols/protocol_tomo_extraction.py [hunk 3]
```diff
@@ -24,18 +24,17 @@
...
 import pyworkflow.protocol.params as params
-
 from tomo.protocols import ProtTomoBase
 from tomo.objects import SetOfSubTomograms, SubTomogram
-
 import eman2
 from eman2.constants import *
 
-# Micrograph type constants for particle extraction
+import xmippLib
```
[the remark this answers is not in the document]
Comment (adrianq) [reply eman2/protocols/protocol_tomo_extraction.py]: For the record, Estrella just talked to @ponesa. He is OK with this change but would like to know @delarosatrevin opinion about this...

[the remark this answers is not in the document]
Comment (azazellochg) [reply eman2/protocols/protocol_tomo_extraction.py]: @adrianq I think this is a very bad idea to import xmipplib. Image handler class can get you dimensions, scale and read/write images. It still uses xmipp, of course, but it's better to have an abstraction layer so in future we won't depend on xmipp.

[the remark this answers is not in the document]
Comment (delarosatrevin) [reply eman2/protocols/protocol_tomo_extraction.py]: I agree with @azazellochg here. It is better to use the ImageHandler and similar classes. As he said, it end up using Xmipp underneath, but it is centralized in this way.

[the remark this answers is not in the document]
Comment (pconesa) [reply eman2/protocols/protocol_tomo_extraction.py]: I agree, but ImageHandler does not have a scale with splines, only scale fourier.

We can add it to devel (or scipion 3) but meanwhile, unless we wait for scipion 3 to release tomo code (I'm not in favour)....we need to hack something.

We "extend" the imageHandler, in the same way we have extended the EMProtocol to have createSetOfTomograms and so on. @delarosatrevin ?

[the remark this answers is not in the document]
Comment (adrianq) [reply eman2/protocols/protocol_tomo_extraction.py]: For future reference, this is how this issue got solved https://github.com/scipion-em/scipion-em-tomo/pull/18/commits/8955520c5287db4dcbb8b370a042a7d1f2a17688

**Pull request merged**